pilot/pkg/xds/discovery.go

```diff
@@ -294,7 +294,15 @@ func (s *DiscoveryServer) globalPushContext() *model.PushContext {
 
 // ConfigUpdate implements ConfigUpdater interface, used to request pushes.
 func (s *DiscoveryServer) ConfigUpdate(req *model.PushRequest) {
-	if len(model.ConfigsOfKind(req.ConfigsUpdated, kind.Address)) > 0 {
+	if features.EnableUnsafeAssertions {
+		// This operation is really slow, which makes tests fail for unrelated reasons, so we process it async.
+		go func() {
```

Comment:
This `go` and comment is copied from something _really_ slow. Like it takes ~1s per xds object. HasConfigsOfKind should be very very quick. Making a goroutine for it is a bit excessive?

Reply:
I did copy this, because i was afraid this will make test flake. Havn't tried though